contrib/flake.nix

```diff
@@ -3,93 +3,91 @@
-      };
-    in {
-      nvim = mkApp self.packages."${system}".neovim;
-      nvim-debug = mkApp self.packages."${system}".neovim-debug;
-    };
-
-    defaultApp."${system}" = self.apps."${system}".nvim;
-
-    devShell."${system}" = self.packages."${system}".neovim-developer;
-  };
+      rec {
+
+        packages = flake-utils.lib.flattenTree rec {
+
+          neovim = pkgs.neovim-unwrapped.overrideAttrs (oa: {
+            version = "master";
+            src = ../.;
+
+            buildInputs = oa.buildInputs ++ ([
+              pkgs.tree-sitter
+            ]);
+
+            cmakeFlags = oa.cmakeFlags ++ [
+              "-DUSE_BUNDLED=OFF"
+            ];
+          });
+
+          # a development binary to help debug issues
+          neovim-debug = (neovim.override {
+            stdenv = pkgs.llvmPackages_latest.stdenv;
+            lua = pkgs.enableDebugging pkgs.luajit;
+          }).overrideAttrs (oa: {
+            cmakeBuildType = "Debug";
+            cmakeFlags = oa.cmakeFlags ++ [
+              "-DMIN_LOG_LEVEL=0"
+            ];
+          });
+
+          # for neovim developers, very slow
+          # brings development tools as well
+          neovim-developer =
+            let
+              lib = nixpkgs.lib;
+              pythonEnv = pkgs.python3;
+              luacheck = pkgs.luaPackages.luacheck;
+            in
+            (neovim-debug.override ({ doCheck = true; })).overrideAttrs (oa: {
+              cmakeFlags = oa.cmakeFlags ++ [
+                "-DLUACHECK_PRG=${luacheck}/bin/luacheck"
+                "-DMIN_LOG_LEVEL=0"
+                "-DENABLE_LTO=OFF"
+                "-DUSE_BUNDLED=OFF"
+                # https://github.com/google/sanitizers/wiki/AddressSanitizerFlags
+                # https://clang.llvm.org/docs/AddressSanitizer.html#symbolizing-the-reports
+                "-DCLANG_ASAN_UBSAN=ON"
+              ];
+
+              nativeBuildInputs = oa.nativeBuildInputs ++ (with pkgs; [
+                pythonEnv
+                include-what-you-use # for scripts/check-includes.py
+                jq # jq for scripts/vim-patch.sh -r
+                doxygen
+              ]);
+
+              shellHook = oa.shellHook + ''
+                export NVIM_PYTHON_LOG_LEVEL=DEBUG
+                export NVIM_LOG_FILE=/tmp/nvim.log
+
+                export ASAN_OPTIONS="log_path=./test.log:abort_on_error=1"
+                export UBSAN_OPTIONS=print_stacktrace=1
+              '';
+            });
+        };
+
+        defaultPackage = packages.neovim;
+
+        apps = {
+          nvim = flake-utils.lib.mkApp { drv = packages.neovim; name = "nvim"; };
+          nvim-debug = flake-utils.lib.mkApp { drv = packages.neovim-debug; name = "nvim"; };
+        };
+
+        defaultApp = apps.nvim;
+
+      }
+    );
+    # overlay = final: prev: {
+    #   inherit (self.packages."${system}") neovim neovim-debug;
+    # };
+    # devShell."${system}" = self.packages."${system}".neovim-developer;
```

Comment:
why remove my devShell :'( ?

Reply:
Don't worry it's coming back! That's why I marked it a draft